packages/std/src/math/uint128.rs

```diff
@@ -205,119 +214,167 @@ impl From<Uint128> for u128 {
 
 impl fmt::Display for Uint128 {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "{}", self.0)
+        self.0.fmt(f)
     }
 }
 
 impl ops::Add<Uint128> for Uint128 {
     type Output = Self;
 
     fn add(self, rhs: Self) -> Self {
-        Uint128(self.u128().checked_add(rhs.u128()).unwrap())
+        Uint128(
+            self.u128()
+                .checked_add(rhs.u128())
+                .unwrap_or_else(|| panic!("attempt to add with overflow")),
+        )
     }
 }
 
 impl<'a> ops::Add<&'a Uint128> for Uint128 {
     type Output = Self;
 
     fn add(self, rhs: &'a Uint128) -> Self {
-        Uint128(self.u128().checked_add(rhs.u128()).unwrap())
+        self + *rhs
     }
 }
 
 impl ops::Sub<Uint128> for Uint128 {
     type Output = Self;
 
     fn sub(self, rhs: Self) -> Self {
-        Uint128(self.u128().checked_sub(rhs.u128()).unwrap())
+        Uint128(
+            self.u128()
+                .checked_sub(rhs.u128())
+                .unwrap_or_else(|| panic!("attempt to subtract with overflow")),
+        )
     }
 }
 
 impl<'a> ops::Sub<&'a Uint128> for Uint128 {
     type Output = Self;
 
     fn sub(self, rhs: &'a Uint128) -> Self {
-        Uint128(self.u128().checked_sub(rhs.u128()).unwrap())
+        self - *rhs
+    }
+}
+
+impl ops::Mul<Uint128> for Uint128 {
+    type Output = Self;
+
+    fn mul(self, rhs: Self) -> Self::Output {
+        Self(
+            self.u128()
+                .checked_mul(rhs.u128())
+                .unwrap_or_else(|| panic!("attempt to multiply with overflow")),
+        )
+    }
+}
+
+impl<'a> ops::Mul<&'a Uint128> for Uint128 {
+    type Output = Self;
+
+    fn mul(self, rhs: &'a Uint128) -> Self::Output {
+        self * *rhs
```

Comment:
What about using the function names here for the cases where we only deref? Seems a bit more obvious to me.

```suggestion
        self.mul(*rhs)
```

Reply:
I don't really have a strong opinion. Sure.